applications/crossbar/src/cb_mobile_manager.erl

```diff
@@ -0,0 +1,94 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2010-2015, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(cb_mobile_manager).
+
+-export([delete_account/1]).
+
+-include("./crossbar.hrl").
+
+-define(MOD_CONFIG_CAT, <<"mobile_manager">>).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+delete_account(Context) ->
+    AccountId = cb_context:account_id(Context),
+    AuthToken = cb_context:auth_token(Context),
+
+    UrlString = req_uri([<<"accounts">>, AccountId]),
+    Headers = req_headers(AuthToken),
+
+    lager:debug("mobile_manager delete via ~s", [UrlString]),
+
+    Resp = ibrowse:send_req(UrlString, Headers, 'delete'),
+    handle_resp(Resp).
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_resp(ibrowse_ret()) -> 'ok'.
+handle_resp({'ok', "200", _, Resp}) ->
+    lager:debug("mobile_manager success ~s", [decode(Resp)]);
+handle_resp({'ok', Code, _, Resp}) ->
+    lager:warning("mobile_manager error ~p. ~s", [Code, decode(Resp)]);
+handle_resp(_Error) ->
+    lager:error("mobile_manager fatal error ~p", [_Error]).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec decode(string()) -> ne_binary().
+decode(JSON) ->
+    try wh_json:encode(JSON) of
+        JObj -> wh_json:decode(JObj)
+    catch
+        'error':_R ->
+            io:format("~p~n", [_R]),
+            JSON
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec req_uri(ne_binaries()) -> list().
+req_uri(ExplodedPath) ->
+    Url = whapps_config:get_binary(?MOD_CONFIG_CAT, <<"url">>),
+    Uri = wh_util:uri(Url, ExplodedPath),
+    binary:bin_to_list(Uri).
```

Comment:
My guess is that this way you know it turns a binary into a list. 
Whereas with a random call to to_binary you have no idea of the type you're turning into a list. 
Easier to read / refactor, sometimes. 
I'm sure that this was the OTP team's idea when the designed this API. 
(Actually not exactly, as I feel like they never really had a discussion for some places in the stdlib)

> On 20 Nov 2015, at 18:15, James Aimonetti <notifications@github.com> wrote:
> 
> In applications/crossbar/src/cb_mobile_manager.erl:
> 
> > +        'error':_R ->
> > +            io:format("~p~n", [_R]),
> > +            JSON
> > +    end.
> > +
> > +%%--------------------------------------------------------------------
> > +%% @private
> > +%% @doc
> > +%%
> > +%% @end
> > +%%--------------------------------------------------------------------
> > +-spec req_uri(ne_binaries()) -> list().
> > +req_uri(ExplodedPath) ->
> > +    Url = whapps_config:get_binary(?MOD_CONFIG_CAT, <<"url">>),
> > +    Uri = wh_util:uri(Url, ExplodedPath),
> > +    binary:bin_to_list(Uri).
> Why use the binary module vs wh_util:to_binary/1 which will wrap that up? Seems non-standard with the rest of kazoo.
> 
> —
> Reply to this email directly or view it on GitHub.
>